recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,170 @@
+import os
+import copy
+from collections import namedtuple
+from conans import ConanFile, CMake, tools
+
+class SociConan(ConanFile):
+    name = "soci"
+    settings = "os", "arch", "compiler", "build_type"
+    description = "SOCI - The C++ Database Access Library"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://soci.sourceforge.net/"
+    license = "BSL-1.0"
+    generators = "cmake"
+    topics = ("conan", "soci", "database")
+    exports_sources=["CMakeLists.txt", "cmake/SOCI-override.cmake"]
+
+    backends = [
+#     "db2",
+      "empty",
+#     "firebird",
+      "mysql",
+      "odbc",
+#     "oracle",
+      "postgresql",
+      "sqlite3"
+    ]
+
+    options = {
+        "shared": [True, False],
+        "with_boost": [True, False],
+        "with_all_backends": [True, False]
```

Comment:
This backend is a bit weird. I would just drop it to avoid the possibility where `with_all_backend=True` and `with_backend_mysql=False`.
Dropping it will enable you to remove the method `_with_backend`.